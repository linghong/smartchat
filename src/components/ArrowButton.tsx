import { FC } from 'react';

type ButtonProps = {
  disabled: boolean;
};

const ArrowButton: FC<ButtonProps> = ({ disabled = false }) => {
  return (
    <button
      type="submit"
      disabled={disabled}
      aria-label="Send message"
      className={`p-1 border-none opacity-60 hover:stone-600 disabled:cursor-not-allowed`}
    >
      <svg
        viewBox="0 0 18 18"
        className={`w-5 h-5 rotate-90 fill-current`}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path d="M10.894 2.553a1 1 0 00-1.788 0l-7 14a1 1 0 001.169 1.409l5-1.429A1 1 0 009 15.571V11a1 1 0 112 0v4.571a1 1 0 00.725.962l5 1.428a1 1 0 001.17-1.408l-7-14z">
          <title>Arrow Submit Button</title>
        </path>
      </svg>
    </button>
  );
};
export default ArrowButton;
