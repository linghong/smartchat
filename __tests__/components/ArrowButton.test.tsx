import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import ArrowButton from '@/src/components/ArrowButton'

describe('ArrowButton Component', () => {

  test('has correct attributes when enabled', () => {
    const { getByRole } = render(<ArrowButton disabled={false} />)
    const button = getByRole('button')

    expect(button).not.toHaveAttribute('disabled')
    expect(button).toHaveClass('opacity-60')
    expect(button).toHaveClass('hover:stone-600')
    expect(button).toHaveClass('disabled:cursor-not-allowed')
  })

  test('has correct attributes when disabled', () => {
    const { getByRole } = render(<ArrowButton disabled={true} />)
    const button = getByRole('button')
    expect(button).toHaveClass('opacity-60')
    expect(button).toHaveClass('hover:stone-600')
    expect(button).toHaveClass('disabled:cursor-not-allowed')
  })

  test('renders SVG with correct attributes', () => {
    const { container } = render(<ArrowButton disabled={false} />)
    const svg = container.querySelector('svg')
    const title = container.querySelector('svg title');

    expect(svg).toHaveAttribute('viewBox', '0 0 18 18')
    expect(title).toHaveTextContent('Arrow Submit Button')
  })

  test('renders correctly and matches the snapshot when enabled', () => {
    const { asFragment } = render(<ArrowButton disabled={false} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
