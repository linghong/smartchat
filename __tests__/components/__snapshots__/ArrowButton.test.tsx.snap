// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowButton Component renders correctly and matches the snapshot when disabled 1`] = `
<DocumentFragment>
  <button
    class="p-1 border-none opacity-60 hover:stone-600 disabled:cursor-not-allowed"
    disabled=""
    type="submit"
  >
    <svg
      class="w-5 h-5 rotate-90 fill-current"
      viewBox="0 0 18 18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.894 2.553a1 1 0 00-1.788 0l-7 14a1 1 0 001.169 1.409l5-1.429A1 1 0 009 15.571V11a1 1 0 112 0v4.571a1 1 0 00.725.962l5 1.428a1 1 0 001.17-1.408l-7-14z"
      >
        <title>
          Arrow Submit Button
        </title>
      </path>
    </svg>
  </button>
</DocumentFragment>
`;

exports[`ArrowButton Component renders correctly and matches the snapshot when enabled 1`] = `
<DocumentFragment>
  <button
    class="p-1 border-none opacity-60 hover:stone-600 disabled:cursor-not-allowed"
    type="submit"
  >
    <svg
      class="w-5 h-5 rotate-90 fill-current"
      viewBox="0 0 18 18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.894 2.553a1 1 0 00-1.788 0l-7 14a1 1 0 001.169 1.409l5-1.429A1 1 0 009 15.571V11a1 1 0 112 0v4.571a1 1 0 00.725.962l5 1.428a1 1 0 001.17-1.408l-7-14z"
      >
        <title>
          Arrow Submit Button
        </title>
      </path>
    </svg>
  </button>
</DocumentFragment>
`;
